### PR TITLE
[scala][akka] Update dependencies for scala 2.13 in scala-akka

### PR DESCRIPTION
**Pull request by ex0ns**
The generated files for the scala-akka binding are not compatible with scala 2.13
This issue was raised in https://github.com/OpenAPITools/openapi-generator/issues/8552

I used the versions lised on this issues, but I think that we could event use more recent version of akka/akka_http, what do you think ?

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (I'm not sure that this is the correct branch, might be 5.1 instead)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Could you please review this

@clasnake (2017/07), @jimschubert (2017/09) ❤️, @shijinkui (2018/01), @ramzimaalej (2018/03), @chameleon82 (2020/03), @Bouillie (2020/04)

### CODE REVIEW

**Comment by ex0ns** [comment]
We need to implement the support for multiple scala version inside `maven` or switch to sbt as `scala-sttp` does.
I tried to implement the first version by having multiple passes in the `pom.xml` but there is a strange behavior when generating the template, changes to `modules/openapi-generator/src/main/resources/scala-akka-client/pom.mustache` are not visible in the generated ` samples/client/petstore/scala-akka/pom.xml` after running `./bin/generate-samples.sh bin/configs/scala*` 

Is there a special processing for the `pom.mustache` file ?

**Comment by wing328** [comment]
> could you please clarify why this was added ? Thanks

Could be due to some manual changes in pom.xml to run the integration tests (used in the CI)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java [hunk 1]
```diff
@@ -166,7 +166,6 @@ public void processOpts() {
         additionalProperties.put(CodegenConstants.INVOKER_PACKAGE, invokerPackage);
 
         supportingFiles.add(new SupportingFile("README.mustache", "", "README.md"));
-        supportingFiles.add(new SupportingFile("pom.mustache", "", "pom.xml"));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:169]: @ex0ns should this change be reverted?

**Comment by ex0ns** [comment]
I'll come back to this pull request soon, but I think that in order to support compilation for  multiple scala's versions, we will have to use sbt here and replace the `pom.xml` content by some generic content (such as the one in the sttp generator).

I have successfully used the new template in a 2.13 project, but I'm still facing some issues regarding the compatibility with 2.12

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java (hunk 1 above)
Comment (ex0ns) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:169]: I'm not sure, the goal is to compile for both 2.13 and 2.12 which is not possible/easy with maven.
I migrated this codegen to sbt.

**Comment by ex0ns** [comment]
I am finally back with an updated version that uses akka 2.6 and that fixes a lot of warning with 2.13.
The codegen seems to be working fine:
```sh
./bin/generate-samples.sh ./bin/configs/scala-akka.yaml
```
And then run sbt in the generated `samples/client/petstore/scala-akka`

```sh
sbt '+compile'
```

I'm however not able to successfully run `sbt test` in this folder, but I might be missing something (because even on the source branch those tests are not running for me)

File: samples/client/petstore/scala-akka/src/test/scala/PetApiTest.scala [hunk 2]
```diff
@@ -118,6 +119,5 @@ class PetApiTest extends AsyncFlatSpec with Matchers {
         }
       }
   }
-  */
+   */
```
Comment (ramzimaalej) [inline samples/client/petstore/scala-akka/src/test/scala/PetApiTest.scala:122]: Can you remove this commented out block as well please?

**Review by ramzimaalej — CHANGES_REQUESTED** [review]
One minor comment in addition to fixing the broken test.

File: samples/client/petstore/scala-akka/src/test/scala/PetApiTest.scala (hunk 2 above)
Comment (ex0ns) [reply samples/client/petstore/scala-akka/src/test/scala/PetApiTest.scala:122]: That is a good catch, thanks. 
On the other hand I was not able to reproduce the CircleCI bug and I would really like some help on that, I'm not sure to understand what the `circe_parallel` script is supposed to do, but I can't run it locally.

**Comment by ex0ns** [comment]
This is ready to be review, could someone take a look ?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:169]: the next version is a patch release so removing this file seems to be a breaking change.

can we simply keep it and update the dependency versions?

Comment (ex0ns) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java:169]: I've added the pom.xml back (as well as the pom.mustache) :ok_hand:

**Comment by wing328** [comment]
Tested locally and the output compiles without issues.

**Pull request merged**